### PR TITLE
Add grpc upto 1.39.0 and update dependencies

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by adamjstewart** [comment]
Do you have any interest in updating the `py-grpcio` package too? Last I recall I was having a lot of build issues with it.

**Comment by iarspider** [comment]
@adamjstewart I can look into it.

File: var/spack/repos/builtin/packages/grpc/package.py [hunk 1]
```diff
@@ -11,11 +11,16 @@ class Grpc(CMakePackage):
...
+    version('1.39.0', sha256='b16992aa1c949c10d5d5ce2a62f9d99fa7de77da2943e643fb66dcaf075826d6')
+    version('1.38.1', sha256='f60e5b112913bf776a22c16a3053cc02cf55e60bf27a959fd54d7aaf8e2da6e8')
+    version('1.38.0', sha256='abd9e52c69000f2c051761cfa1f12d52d8b7647b6c66828a91d462e796f2aede')
+    version('1.37.1', sha256='acf247ec3a52edaee5dee28644a4e485c5e5badf46bdb24a80ca1d76cb8f1174')
+    version('1.37.0', sha256='c2dc8e876ea12052d6dd16704492fd8921df8c6d38c70c4708da332cf116df22')
+    version('1.36.4', sha256='8eb9d86649c4d4a7df790226df28f081b97a62bf12c5c5fe9b5d31a29cd6541a')
+    version('1.36.3', sha256='bb6de0544adddd54662ba1c314eff974e84c955c39204a4a2b733ccd990354b7')
     version('1.33.1', sha256='58eaee5c0f1bd0b92ebe1fa0606ec8f14798500620e7444726afcaf65041cb63')
-    version('1.32.0', sha256='f880ebeb2ccf0e47721526c10dd97469200e40b5f101a0d9774eb69efa0bd07a')
-    version('1.31.0', sha256='1236514199d3deb111a6dd7f6092f67617cd2b147f7eda7adbafccea95de7381')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/grpc/package.py]: Why these two versions have been removed?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/grpc/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/grpc/package.py]: Copy-paste error. I will restore them.

**Review by alalazo — APPROVED** [review]

**Pull request merged**